posthog/queries/retention/retention_events_query.py

```diff
@@ -121,9 +135,7 @@ def get_query(self) -> Tuple[str, Dict[str, Any]]:
 
         prop_query, prop_params = self._get_prop_groups(
             self._filter.property_groups,
-            person_properties_mode=PersonPropertiesMode.DIRECT_ON_EVENTS
-            if self._using_person_on_events
-            else PersonPropertiesMode.USING_PERSON_PROPERTIES_COLUMN,
+            person_properties_mode=self._get_person_properties_mode(),
             person_id_joined_alias=f"{self.EVENT_TABLE_ALIAS if self._using_person_on_events else self.DISTINCT_ID_TABLE_ALIAS}.person_id",
```

Comment:
Didn't look too deeply, but this seems like it will break? 

We now need to be very careful of which identifier we're using for the person_id, since there's 3:

1. The `e.person_id`
2. The `pdi.person_id` (shouldn't exist when person on events v1|v2 is enabled.
3. The `override.person_id`: main person_id to use, if exists.

I'd suggest creating a `main_person_id` field (or, to keep inline with other conventions, just call it `aggregation_target`) and passing that around, which coalesces the above appropriately, so nothing else has to worry about the 'correct' personID coming in.

Reply:
Same on L158, which is going to `e.person_id`

Reply:
Ah, I see below you're doing this correctly for `target`. I think it's a bit too late there though, needs to go in beforehand, so it can feed into things downstream more easily.

And then, the target function will become pretty simple too.

Reply:
yeah actually this joined alias might get complicated, need to figure out the best approach. Also we need to kind of do this `if(notEmpty(overrides.person_id),overrides.person_id,e.person_id)` everywhere which is annoying

Reply:
That's what I mean by coalescing:

`if(notEmpty(overrides.person_id), overrides.person_id, e.person_id) as aggregation_target`, and everywhere that depends on person_id_alias gets passed in aggregation_target